mariamihaelafainarea/assignment3/Assignment3

```diff
@@ -0,0 +1,192 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package assignment3;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Set;
+import java.util.TreeSet;
+
+/**
+ *
+ * @author maria
+ */
+public class Assignment3 {
+    //dictionary of the words and the grid
+    private char[][] grid;
+    private MyDictionary dictionary;
+    private int n,m;
+
+    public MyDictionary getDictionary() {
+        return dictionary;
+    }
+
+    public void setDictionary(MyDictionary dictionary) {
```

Comment:
You never set either dictionary or grid after in the constructor. Maybe remove the setters and make all those variables final: that is a good way to show a guarantee that every instance of this class is formed to solve a specific dictionary and grid.

Reply:
done